sophia/src/dataset/test.rs

```diff
@@ -91,7 +91,13 @@ pub fn populate_nodes_types<D: MutableDataset>(d: &mut D) -> MDResult<D, ()> {
     Ok(())
 }
 
-pub fn as_box_q<Q: Quad>(quad: Q) -> ([BoxTerm; 3], Option<BoxTerm>) {
+pub fn as_box_q<Q: Quad, E>(quad: Result<Q, E>) -> ([BoxTerm; 3], Option<BoxTerm>) 
+{
+    let quad = match quad {
+        Ok(q) => q,
+        Err(_) => panic!("as_box_q received an error")
```

Comment:
As a little improvement I would suggest:
```rust
Err(e) => panic!("as_box_q received an error: {}", e)
```

Reply:
this would require that `E` implement `Display` (or `Debug` if we replaced `{}` with `{:?}`, which would probably be better).

In fact, this `panic!` was added by @BruJu to keep the function as general as possible, i.e. *not* constraining `E` to implement `Debug`. Otherwise, the whole `match` could have been replaced with a simple `unwrap()`.

Granted, error types should always implement `Debug`, so this care for genericity might be overzealous... But this is just a test function.